XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs

```diff
@@ -175,9 +177,19 @@ public override void PageLeave(PageLoadedDirection direction, ref bool cancel)
         {
             if (direction == PageLoadedDirection.Forward)
             {
+                //check if we are installing update user sees in textbox
+                if (Path.GetFileNameWithoutExtension(unzippedUpdateFilePath) != Path.GetFileNameWithoutExtension(fileNameTextBox.Text))

```

Comment:
We should only go on on this path if the file is a zip.
Also, we should clear unzippedUpdateFilePath if the slected file is not a zip, othewise we need to add the check for zip extension on line 180 below.